libsolidity/parsing/Parser.cpp

```diff
@@ -1606,7 +1606,24 @@ ASTPointer<Expression> Parser::parseBinaryExpression(
 		{
 			Token op = m_scanner->currentToken();
 			m_scanner->next();
-			ASTPointer<Expression> right = parseBinaryExpression(precedence + 1);
+
+			constexpr unsigned expPrecedence =
```

Comment:
Why can't you use TokenTraits::precedence for this?

Reply:
`TokenTraits::precedence` is not constexpr, but let me see if we can make it constexpr.